src/utils/subscriberBreaches.ts

```diff
@@ -77,16 +78,30 @@ export async function getSubBreaches(
       const filteredBreachDataClasses: string[] = filterBreachDataTypes(
         breach.DataClasses
       );
+      // `allBreaches` is generally the return value of `getBreaches`, which
+      // either loads breaches from the database, or fetches them from the HIBP
+      // API. In the former csae, `AddedDate`, `BreachDate` and `ModifiedDate`
+      // are Date objects, but in the latter case, they are ISO 8601 date
+      // strings. Thus, we normalise that to always be a Date object.
       const subscriberBreach: SubscriberBreach = {
         id: breach.Id,
-        addedDate: breach.AddedDate,
-        breachDate: breach.BreachDate,
+        addedDate:
+          typeof breach.AddedDate === "string"
+            ? parseIso8601Datetime(breach.AddedDate)
+            : breach.AddedDate,
+        breachDate:
+          typeof breach.BreachDate === "string"
+            ? parseIso8601Datetime(breach.BreachDate)
+            : breach.BreachDate,
```

Comment:
I just figured it was nice to avoid _every_ call site of this function having to add a branch to deal with two potentially different types of data, and just do it once and for all here.

Reply:
Yeah good point, done in df314241f.